src/hooks/useMonthHeaders.ts

```diff
@@ -0,0 +1,27 @@
+import { useCallback, useMemo } from 'react';
+import { getMonthCount } from 'src/components/Reports/MPGAIncomeExpensesReport/Helper/getMonthCount';
+import { getMonthInfo } from 'src/components/Reports/MPGAIncomeExpensesReport/Helper/getMonthInfo';
+
+export const useMonthHeaders = (
+  months: string[],
+  colors: { first: string; second: string },
+) => {
+  const monthCount = useMemo(() => {
+    return getMonthCount(months);
+  }, [months]);
+
+  const firstMonth = useMemo(() => getMonthInfo(months), [months]);
+
+  const getBorderColor = useCallback(
+    (index: number) => {
+      return index === 0 ? colors.first : colors.second;
+    },
+    [firstMonth, colors, months],
```

Comment:
[Nice-to-have] Unused dependency in useCallback. Evidence: `firstMonth` in dependency array but not used in function. Impact: Unnecessary re-creation. Suggestion: Remove from dependency array. [Category: File]